### PR TITLE
INT: Introduce expand dependency specification intention

### LINKED ISSUE #4869: Suggestion - quick action to expand dependency specification in Cargo.toml
When adding dependencies to my `Cargo.toml`, it's handy that the latest version is provided as a completion. However, it's still a bit annoying to type out the dependency specifications when the dependency needs to be configured with additional options, like `default_features = false` or `features = [ "serde" ]`. It would be handy to have an action that expands the shorter syntax to the full syntax, placing the cursor before the closing bracket so additional options can be added - replacing `serde = "1.0"` with `serde = { version = "1.0" }`

### PR DESCRIPTION
**Pull request by avrong**
Related to #4869

### TODO
- [x] Provide TOML intention test-base to perform tests correctly

changelog: Add intention to expand dependency specification in `Cargo.toml`

### CODE REVIEW

File: toml/src/main/kotlin/org/rust/toml/intentions/ExpandDependencySpecificationIntention.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        if (table.header.key?.segments?.lastOrNull()?.isDependencyKey != true) return null
+
+        val value = keyValue.value
+        if (value !is TomlLiteral || value.kind !is TomlLiteralKind.String) return null
+
+        return keyValue
+    }
+
+    override fun invoke(project: Project, editor: Editor, ctx: TomlKeyValue) {
+        val newKeyValue = TomlPsiFactory(project).createKeyValue("${ctx.key.text} = { version = ${ctx.value?.text ?: "\"\""} }")
```
Comment (ortem) [inline toml/src/main/kotlin/org/rust/toml/intentions/ExpandDependencySpecificationIntention.kt:40]: Consider introducing variables for readability:
```Kotlin
val crateName = ctx.key.text
val crateVersion = ctx.value?.text ?: "\"\""
```

File: src/main/resources/intentionDescriptions/ExpandDependencySpecificationIntention/description.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<html>
+<body>
+This intention expands dependency description in Cargo.toml manifest file.
```
Comment (ortem) [inline src/main/resources/intentionDescriptions/ExpandDependencySpecificationIntention/description.html:3]:
```suggestion
This intention expands dependency specification in Cargo.toml
```

**Review by ortem — CHANGES_REQUESTED** [review]

File: toml/src/main/kotlin/org/rust/toml/intentions/ExpandDependencySpecificationIntention.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+class ExpandDependencySpecificationIntention : TomlElementBaseIntentionAction<TomlKeyValue>() {
+    override fun getText() = "Expand dependency specification"
+    override fun getFamilyName(): String = text
+
+    override fun findApplicableContextInternal(project: Project, editor: Editor, element: PsiElement): TomlKeyValue? {
+        if (element.containingFile.name != CargoConstants.MANIFEST_FILE) return null
+
+        val keyValue = element.parentOfType<TomlKeyValue>() ?: return null
+        val table = keyValue.parent as? TomlTable ?: return null
+        if (table.header.key?.segments?.lastOrNull()?.isDependencyKey != true) return null
```
[the remark this answers is not in the document]
Comment (avrong) [reply toml/src/main/kotlin/org/rust/toml/intentions/ExpandDependencySpecificationIntention.kt:31]: There is no need to include table dependencies like `[dependencies.foo]` in the current check

[the remark this answers is not in the document]
Comment (ortem) [reply toml/src/main/kotlin/org/rust/toml/intentions/ExpandDependencySpecificationIntention.kt:31]: I mean this property from `Utils.kt`:
```Kotlin
val TomlTableHeader.isDependencyListHeader: Boolean
       get() = key?.segments?.lastOrNull()?.isDependencyKey == true
```
Looks like `table.header.isDependencyListHeader` would be perfect here, isn't it?

[the remark this answers is not in the document]
Comment (avrong) [reply toml/src/main/kotlin/org/rust/toml/intentions/ExpandDependencySpecificationIntention.kt:31]: Oh, yes. You're right

File: toml/src/main/resources/META-INF/toml-only.xml [hunk 4]
```diff
@@ -32,6 +32,11 @@
                          enabledByDefault="true" level="WARNING"
                          implementationClass="org.rust.toml.inspections.CrateNotFoundInspection"/>
 
+        <intentionAction>
+            <className>org.rust.toml.intentions.ExpandDependencySpecificationIntention</className>
+            <category>Cargo.toml</category>
```
Comment (ortem) [inline toml/src/main/resources/META-INF/toml-only.xml:37]:
```suggestion
            <category>Rust/Cargo.toml</category>
```
`<category>Cargo.toml</category>` creates a new category on the top level but this intention should belong to `Rust` instead

**Review by ortem — CHANGES_REQUESTED** [review]

**Review by ortem — APPROVED** [review]
bors r+

**Pull request merged**